File: 0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.py
class Stack():
    def __init__(self):
        self._a = []
        self._max_size = 0

    def push(self, x: int) -> None:
        self._a.append(x)
        self._max_size = max(self._max_size, len(self._a))

    def pop(self) -> int:
        return self._a.pop() if self._a else None

    def top(self) -> int:
        return self._a[-1] if self._a else None

    def empty(self) -> bool:
        return len(self._a) == 0

    def __len__(self) -> int:
        return len(self._a)

    def space(self) -> int:
        return self._max_size

class QueueUsingStack():
    def __init__(self):
        self.s1 = Stack()
        self.s2 = Stack()

    def push(self, x: int) -> None:
        while not self.s1.empty():
            self.s2.push(self.s1.pop())
        self.s1.push(x)
        while not self.s2.empty():
            self.s1.push(self.s2.pop())

    def pop(self) -> int:
        return self.s1.pop() if not self.s1.empty() else None

    def peek(self) -> int:
        return self.s1.top() if not self.s1.empty() else None

    def empty(self) -> bool:
        return self.s1.empty()

    def __len__(self) -> int:
        return len(self.s1)


if (True):
    MyQueue = QueueUsingStack

File: 0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.java
import java.util.Stack;

class MyQueue {
    private Stack<Integer> pushStack;
    private Stack<Integer> popStack;

    public MyQueue() {
        pushStack = new Stack<>();
        popStack = new Stack<>();
    }

    public void push(int x) {
        pushStack.push(x); // Always push to pushStack
    }

    public int pop() {
        shiftStacks(); // Move elements if needed
        return popStack.pop();
    }

    public int peek() {
        shiftStacks(); // Move elements if needed
        return popStack.peek();
    }

    public boolean empty() {
        return pushStack.isEmpty() && popStack.isEmpty();
    }

    private void shiftStacks() {
        if (popStack.isEmpty()) {
            while (!pushStack.isEmpty()) {
                popStack.push(pushStack.pop()); // Transfer elements to reverse order
            }
        }
    }
}


/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue obj = new MyQueue();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.peek();
 * boolean param_4 = obj.empty();
 */